File: app/src/main/java/tpcreative/co/qrscanner/common/HistorySingleton.java
package tpcreative.co.qrscanner.common;

public class HistorySingleton {
    private static HistorySingleton instance ;
    private SingletonHistoryListener listener;

    public static HistorySingleton getInstance(){
        if (instance==null){
            synchronized (HistorySingleton.class){
                if (instance==null){
                    instance = new HistorySingleton();
                }
            }
        }
        return instance;
    }

    public void setListener(SingletonHistoryListener listener){
        this.listener = listener;
    }

    public void reloadData(){
        if (listener!=null){
            listener.reloadData();
        }
    }

    public interface SingletonHistoryListener{
        void reloadData();
    }

}

File: app/src/main/java/tpcreative/co/qrscanner/common/HistorySingleton.kt
package tpcreative.co.qrscanner.common

class HistorySingleton {
    private var listener: SingletonHistoryListener? = null
    fun setListener(listener: SingletonHistoryListener?) {
        this.listener = listener
    }

    fun reloadData() {
        if (listener != null) {
            if (Utils.isRequestHistoryReload()){
                listener?.reloadData()
                Utils.setRequestHistoryReload(false)
            }
        }
    }

    fun reloadDataChangeDesign() {
        if (listener != null) {
            listener?.reloadData()
        }
    }

    interface SingletonHistoryListener {
        fun reloadData()
    }

    companion object {
        private var instance: HistorySingleton? = null
        fun getInstance(): HistorySingleton? {
            if (instance == null) {
                synchronized(HistorySingleton::class.java) {
                    if (instance == null) {
                        instance = HistorySingleton()
                    }
                }
            }
            return instance
        }
    }
}